tests/lang_callable/macros/tsymbol_must_match.nim

```diff
@@ -57,3 +57,9 @@ replaceName(nskMacro):
   template t() = #[tt.Error
           ^ (SemSymbolKindMismatch)]#
     discard
+
+macro t1(): untyped =
+  let a = genSym(ident = "a")
+  result = newVarStmt(a, newLit 1)
+
+t1()
```

Comment:
don't know what to say, the `transitionRoutineSymKind` introduced there without any document, now fix it will need much requirements.

Reply:
hope someone can help me with this, no idea how to complete the test case like above code.

Reply:
> don't know what to say, the `transitionRoutineSymKind` introduced there without any document, now fix it will need much requirements.

I'm not sure I understand. Do you mean that you don't know how to document the test?

> hope someone can help me with this, no idea how to complete the test case like above code.

What are the things you have trouble with? I can try to help, but I'm not yet sure with what.

Reply:
first I don't know how to make it produce the error then I can write the expected error in comment, second the  
`newVarStmt` need a value node, don't know how to get rid of `newLit`

Reply:
so the error reported when sem routines.

Reply:
Unfortunately, this comment is not resolved yet.

A documentation comment for the test is missing (what exactly is being tested? the tests description applies, but the fact that the test is specifically for the wrong-symbol-in `var`-symbols-slot case should be highlighted).

In addition, the test as it is now no longer acts as a test for the fix, since with the previous logic in `newSymGNode`, the symbol would have been transitioned to a routine symbol, meaning that no range error would have happened.

> first I don't know how to make it produce the error then I can write the expected error in comment

The error happens when a symbol is transplanted into a symbol slot of a non-routine definition and the symbol. However, the error is indeed not reported at the moment.

> second the `newVarStmt` need a value node, don't know how to get rid of `newLit`

My original suggestion would have be to use the same approach as the tests for the routines use above, but `newLit` can also just stay for now.